### PR TITLE
windows_hotfix resource: test whether a Windows HotFix is installed

### PR DESCRIPTION
**Pull request by mattray**
Adds the ability to check if WIndows hotfixes of the form "KB1234567" are installed or not.

Signed-off-by: Matt Ray <matthewhray@gmail.com>

### CODE REVIEW

**Comment by mattray** [comment]
I wasn't sure how to provide an integration test, KBs are specific to different versions of Windows and many get rolled-up into larger service packs. I do have a working integration test for Windows 2012r2.
```ruby
describe hotfix('KB4019215') do
  it { should be_installed }
end

describe hotfix('KB9999999') do
  it { should_not be_installed }
end
```
```shell
$ bundle exec bin/inspec exec . -t winrm://vagrant@127.0.0.1 --password "vagrant" -p 55985

Profile: tests from .
Version: (not specified)
Target:  winrm://vagrant@http://127.0.0.1:55985/wsman:3389

  Hotfix KB4019215
     ✔  should be installed
  Hotfix KB9999999
     ✔  should not be installed

Test Summary: 2 successful, 0 failures, 0 skipped
```

File: test/unit/resources/hotfix_test.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  it 'fail hotfix on ubuntu' do
+    resource = MockLoader.new(:ubuntu1404).load_resource('hotfix', 'KB4019215')
+    _(resource.installed?).must_equal false
+  end
+
+  # windows
+  it 'verify hotfix on windows' do
+    resource = MockLoader.new(:windows).load_resource('hotfix', 'KB4019215')
+    _(resource.installed?).must_equal true
+  end
```
Comment (adamleff) [inline test/unit/resources/hotfix_test.rb]: Can we add a test for a hotfix that doesn't exist, along with the mock command output we should expect?

**Review by adamleff — CHANGES_REQUESTED** [review]
Hi @mattray! Thanks for this, this is a great addition.

Any chance I can convince you to name the resource `windows_hotfix` instead so we don't have to worry about resource name confusion if we ever have a hotfix-like resource for something else?

Re: integration test, one way of doing this might be to have the chef recipe that sets up the test node query for any installed hotfixes and write out an installed hotfix ID to the filesystem, and then the integration test could read from that file. That way we don't have to worry about catering to different OS flavors and potential roll-ups eating individual IDs.

**Comment by mattray** [comment]
@adamleff I upgraded the PR with your feedback on the windows_hotfix name and added the other test. It will take awhile to sort the integration test, since it's my weekend here in 🇦🇺

**Comment by adamleff** [comment]
Thanks @mattray! The changes look good though you've got some failing Rubocop tests (sorry).

Let me know next week about the integration test progress. If it's going to be a while, we can merge this in and follow-up on the integration tests later.

**Comment by mattray** [comment]
Undid the change that made Rubocop unhappy.

**Comment by mattray** [comment]
I'm working on an integration test based on querying the existing KBs applied and then checking with the ```windows_hotfix``` resource. If you merge this I'll have it ready in the next day or two in a separate PR so I don't have to keep rebasing this one.

**Comment by mattray** [comment]
I've added an integration test that works against my vagrant-based Windows 2012r2.

**Review by adamleff — APPROVED** [review]
This looks great, @mattray. Thank you!

**Review by arlimus — APPROVED** [review]
This is a great addition, thank you @mattray !!

**Pull request merged**